x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/select_inference_id.tsx

```diff
@@ -135,14 +140,34 @@ export const SelectInferenceId = ({
         setIsInferenceFlyoutVisible(!isInferenceFlyoutVisible);
         setIsCreateInferenceApiLoading(false);
         setInferenceAddError(undefined);
+
+        const trainedModelId =
+          'model_id' in modelConfig.service_settings &&
+          (modelConfig.service_settings.model_id === Service.elser ||
+            modelConfig.service_settings.model_id === Service.elasticsearch)
+            ? modelConfig.service_settings.model_id
+            : undefined;
```

Comment:
This logic will be moved, when create inference endpoint is made as an async task, added here to keep the current logic neat